coordinator/src/services/communication/tracker/tracker-response-builder.ts

```diff
@@ -0,0 +1,80 @@
+import {hl7V3} from "@models"
+import pino from "pino"
+import {SpineDirectResponse} from "../../../../../models/spine"
+import {extractHl7v3PrescriptionFromMessage, extractSpineErrorDescription} from "./spine-response-parser"
+
+enum TrackerErrorString {
+  FAILED_TRACKER_REQUEST = "Failed to retrieve prescription from Spine",
+  FAILED_PRESCRIPTION_EXTRACT = "Failed to extract prescription from Spine response",
+  FAILED_PRESCRIPTION_VERIFY = "Failed to verify prescription from Spine",
+  FAILED_ERROR_RESPONSE_EXTRACT = "Could not extract error details from Spine response"
+}
+
+interface TrackerError {
+  errorMessage: string
+  errorDetails?: Array<string>
+}
+
+interface TrackerResponse {
+  statusCode: number
+  prescription?: hl7V3.ParentPrescription
+  error?: TrackerError
+}
+
+const createTrackerError = (error: TrackerErrorString, details?: string | Array<string>): TrackerError => {
+  const messageDetails = details
+    ? Array.isArray(details) ? details : [details]
+    : []
+
+  return {
+    errorMessage: error,
+    errorDetails: messageDetails
+  }
+}
+
+const extractPrescription = (responseBody: string, logger: pino.Logger): hl7V3.ParentPrescription => {
+  try {
+    return extractHl7v3PrescriptionFromMessage(responseBody, logger)
+  } catch (error) {
+    logger.error(`${TrackerErrorString.FAILED_PRESCRIPTION_EXTRACT}: ${error}`)
+  }
+}
+
+const tryExtractErrorMessage = (responseBody: string, logger: pino.Logger): string => {
+  try {
+    return extractSpineErrorDescription(responseBody)
+  } catch (error) {
+    logger.error(`${TrackerErrorString.FAILED_ERROR_RESPONSE_EXTRACT}: ${error}`)
+  }
+}
+
+const createTrackerResponse = (spineResponse: SpineDirectResponse<string>, logger: pino.Logger): TrackerResponse => {
+  const prescription = extractPrescription(spineResponse.body, logger)
+  if (prescription) {
+    logger.info(`Successfully extracted prescription ${prescription.id}`)
+    return {
+      statusCode: 200,
+      prescription: prescription
+    }
+  }
+
+  // TODO: Check which errors should/shouldn't be passed from Spine to user
+  const failureDescription = tryExtractErrorMessage(spineResponse.body, logger) ?? spineResponse.body
+  const error = createTrackerError(
+    TrackerErrorString.FAILED_PRESCRIPTION_EXTRACT,
+    failureDescription
+  )
+
+  return {
+    statusCode: 500,
+    error
+  }
+}
+
+export {
```

Comment:
can we do normal export style (eg `export const...`)

Reply:
I think it's easier to keep track of exports this way, if we have more than a couple